sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.util.Beta;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Group configuration which will be used in Throughput control.
+ */
+@Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class ThroughputControlGroup {
+    private final ThroughputControlMode controlMode;
+    private final String groupName;
+    private final String id;
+    private final boolean isDefault;
+    private final CosmosAsyncContainer targetContainer;
+    private final Integer targetThroughput;
+    private final Double targetThroughputThreshold;
+
+    ThroughputControlGroup(
+        String groupName,
+        CosmosAsyncContainer targetContainer,
+        Integer targetThroughput,
+        Double targetThroughputThreshold,
+        ThroughputControlMode controlMode,
+        boolean isDefault) {
+
+        checkArgument(StringUtils.isNotEmpty(groupName), "Group name can not be null or empty");
+        checkNotNull(targetContainer, "Target container can not be null");
+        checkArgument(targetThroughput == null || targetThroughput > 0, "Target throughput should be greater than 0");
+        checkArgument(
+            targetThroughputThreshold == null || (targetThroughputThreshold > 0 && targetThroughputThreshold <= 1),
+            "Target throughput threshold should between (0, 1]");
+
+        this.groupName = groupName;
+        this.targetContainer = targetContainer;
+        this.targetThroughput = targetThroughput;
+        this.targetThroughputThreshold = targetThroughputThreshold;
+        this.controlMode = controlMode;
+        this.isDefault = isDefault;
+
+        this.id = this.getId();
+    }
+
+    /**
+     * Get the throughput control group name.
+     *
+     * @return the group name.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public String getGroupName() {
+        return this.groupName;
+    }
+
+    /**
+     * Get the throughput control group target container.
+     *
+     * @return the {@link CosmosAsyncContainer}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosAsyncContainer getTargetContainer() {
```

Comment:
we shouldn't expose this as public API.

This ties the ThroughputControlGroup to the async api. what if the customer is using sync API?

please hide this api.

Reply:
Yup, make sense, hide it now